llvm/include/llvm/CAS/FileSystemCache.h

```diff
@@ -126,9 +126,10 @@ class FileSystemCache : public ThreadSafeRefCountedBase<FileSystemCache> {
 
   /// Request a directory entry. The first parameter is the parent to look
   /// under, the second is the name of the entry, and the third is true if the
-  /// name came from a call to \a PreloadTreePathType.
-  using RequestDirectoryEntryType =
-      function_ref<Expected<DirectoryEntry *>(DirectoryEntry &, StringRef)>;
+  /// name is requested because a prior result from PreloadTreePath failed to
+  /// match at this position.
+  using RequestDirectoryEntryType = function_ref<Expected<DirectoryEntry *>(
```

Comment:
I'm not thrilled about this API, but it still allows the user of FileSystemCache to either provide preload realpath and honour this flag, or to only provide request directory and decide internally how to handle case-insensitivity.

I originally wanted to handle this inside preload realpath itself or by returning more information from it, but that seemed tricky since the case-insensitivity may be inside a path only seen because of symlink traversal, and it's not clear how to detect which component of the path it happened in without either giving up on non-ascii case-insensitive paths so that we can check for the casing directly, or calling realpath on many components of the path.